> ## `nixpkgs-review` result
> 
> Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).
> 
> Command: `nixpkgs-review pr 383665`
> ### `x86_64-linux`
> ❌ 1 package failed to build:

```
> ok         github.com/emitter-io/emitter/internal/event/crdt       0.030s
       > ok      github.com/emitter-io/emitter/internal/message  0.307s
       > --- FAIL: TestNewClient (0.00s)
       >     http_test.go:50:
       >          Error Trace:    /build/source/internal/network/http/http_test.go:50
       >            Error:          Expected value not to be nil.
       >          Test:           TestNewClient
       >     http_test.go:51:
       >            Error Trace:    /build/source/internal/network/http/http_test.go:51
       >            Error:          Received unexpected error:
       >                             lookup google.com on [::1]:53: read udp [::1]:33291->[::1]:53: read: connection refused
       >                Test:           TestNewClient
       >     http_test.go:50:
       >            Error Trace:    /build/source/internal/network/http/http_test.go:50
       >            Error:          Expected value not to be nil.
       >          Test:           TestNewClient
       >     http_test.go:51:
       >            Error Trace:    /build/source/internal/network/http/http_test.go:51
       >            Error:          Received unexpected error:
       >                             lookup google.com on [::1]:53: read udp [::1]:48613->[::1]:53: read: connection refused
       >                Test:           TestNewClient
       > 2025/02/20 15:06:38 http: superfluous response.WriteHeader call from github.com/emitter-io/emitter/internal/network/http.(*handler2).ServeHTTP (http_test.go:124)
       > FAIL
       > FAIL    github.com/emitter-io/emitter/internal/network/http     0.008s
       > FAIL
```

--- a/pkgs/by-name/em/emitter/package.nix
+++ b/pkgs/by-name/em/emitter/package.nix
@@ -1,0 +1,44 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  writableTmpDirAsHomeHook,
+}:
+
+buildGoModule rec {
+  pname = "emitter";
+  version = "3.1";
+
+  src = fetchFromGitHub {
+    owner = "emitter-io";
+    repo = "emitter";
+    tag = "v${version}";
+    hash = "sha256-eWBgRG0mLdiJj1TMSAxYPs+8CqLNaFUOW6/ghDn/zKE=";
+  };
+
+  vendorHash = "sha256-6K9KAvb+05nn2pFuVDiQ9IHZWpm+q01su6pl7CxXxBY=";
+
+  nativeCheckInputs = [ writableTmpDirAsHomeHook ];
+
+  ldflags = [
+    "-X github.com/emitter-io/emitter/internal/command/version.version=${version}"
+    "-X github.com/emitter-io/emitter/internal/command/version.commit=${src.rev}"
+  ];
+
+  doCheck = true;
+
+  checkFlags = [
+    # Tests require network access
+    "-skip=^Test(NewClient|Statsd_BadSnapshot|Statsd_Configure|Join|Random)$"
+  ];
+
+  __darwinAllowLocalNetworking = true;
+
+  meta = {
+    description = "High performance, distributed and low latency publish-subscribe platform";
+    homepage = "https://emitter.io/";
+    license = lib.licenses.agpl3Plus;
+    maintainers = with lib.maintainers; [ sikmir ];
+    mainProgram = "emitter";
+  };
+}

